classes/controller/class-block-search.php

```diff
@@ -0,0 +1,264 @@
+<?php
+/**
+ * Block search
+ *
+ * @package P4BKS\Controllers
+ */
+
+namespace P4GBKS\Controllers;
+
+use wpdb;
+
+/**
+ * Block search in post content
+ *
+ * Usage:
+ *   $search = new Search ( $wpdb );
+ *   // All blocks with a className attribute in block options
+ *   $search->find( 'className' );
+ *   // All blocks of type planet4-blocks/covers
+ *   $search->find( null, ['block_type' => 'planet4-blocks/covers'] );
+ *   // All blocks sorted by post_title
+ *   $search->find( null, null, [ 'post_title' ] );
+ */
+class Block_Search {
+
+	/**
+	 * @var wpdb
+	 */
+	private $db;
+
+	/**
+	 * @var array[]
+	 */
+	private $results = [];
+
+	/**
+	 * @var int
+	 */
+	private $block_count = 0;
+
+	/**
+	 * @var int
+	 */
+	private $post_count = 0;
+
+	/**
+	 * @param wpdb $wpdb WP DB object.
+	 */
+	public function __construct( wpdb $wpdb ) {
+		$this->db = $wpdb;
+	}
+
+	/**
+	 * @param ?string   $search  Search in blocks names and options.
+	 * @param ?array    $filters Filter by exact dimension value.
+	 * @param ?string[] $sort    Sort by dimension (natural sort).
+	 *
+	 * @return array[] Array of parsed items
+	 * Item format:
+	 * [
+	 *   'post_id'       => int,
+	 *   'post_title'    => string,
+	 *   'post_status'   => string,
+	 *   'post_type'     => string,
+	 *   'post_date'     => string (datetime),
+	 *   'post_modified' => string (datetime),
+	 *   'guid'          => string,
+	 *   'block_ns'      => string,
+	 *   'block_type'    => string,
+	 *   'block_opts'    => string (json),
+	 *   'match'         => string,
+	 * ]
+	 */
+	public function find(
+		?string $search = null,
+		?array $filters = null,
+		?array $sort = null
+	): array {
+		$this->reset();
```

Comment:
It's probably better if we create an instance for each search if we're executing multiple searches in the same request. So I guess it doesn't make sense to reset the state of the instance here since it's only supposed to be used once?